install.sh

```diff
@@ -18,15 +18,27 @@ set -euo pipefail
   at_exit() {
     ret=$?
     if [[ $ret -gt 0 ]]; then
-      log "the script failed with error $ret.\n" \
-        "\n" \
-        "To report installation errors, submit an issue to\n" \
-        "    https://github.com/direnv/direnv/issues/new/choose"
+      log "the script failed with error ${ret}.\n" \
+      "To report installation errors, submit an issue to\n" \
+      "https://github.com/direnv/direnv/issues/new/choose"
     fi
     exit "$ret"
   }
   trap at_exit EXIT
 
+  if [ $# -ge 1 ];then
+    if [ "${1}" == "--help" ] || [ "${1}" == "-h" ]; then
+      printf "USAGE: ./install-direnv.sh [bin_path]\n"
+      printf "bin_path=~/.local/bin ./install-direnv.sh\n"
+      exit 0
+    elif [ -d "$(realpath "$1")" ] && [ -w "$(realpath "$1")" ]; then
+      bin_path="$(realpath "$1")"
+    else
+      log "bin_path ${1} does not exist or is not writeable" >&2
```

Comment:
Log already writes to stderr

```suggestion
      log "bin_path ${1} does not exist or is not writeable"
```